fgi/icon.py

```diff
@@ -38,38 +38,59 @@
     "xbox-series-s": "xbox",
 }
 
+platform_text_map = {
+    "playstation2": "PlayStation 2",
+    "playstation3": "PlayStation 3",
+    "playstation4": "PlayStation 4",
+    "playstation5": "PlayStation 5",
+    "psv": "PlayStation Vita",
+    "psp": "PlayStation Portable",
+    "xbox-one": "Xbox One",
+    "xbox-360": "Xbox 360",
+    "xbox-series-x": "Xbox Series X",
+    "xbox-series-s": "Xbox Series S",
+}
+
 class IconFactory:
     def __init__(self, data):
         self.data = data
         self.cache = dict()
 
-    def icon(self, name, fallback="misc-fallback", fallback_html=None):
+    def icon(self, name, disp_name=None, fallback="misc-fallback", fallback_html=None):
         if fallback and name not in self.data:
             name = fallback
         if fallback_html and name not in self.data:
             return fallback_html
 
         if name in self.cache:
-            return self.cache[name]
+            return self.cache[f"{name}{disp_name}"]
```

Comment:
I think `[f"{name}{disp_name if disp_name else ''}"]` is better.
because if disp_name is None, then f"{name}{disp_name}" will be "...None". That will be conflict to disp_name=="None"